### PR TITLE
Big redesign of the main table view

### PR DESCRIPTION
**Pull request by Flight**
* search moved to top
* mobile view improvements
* folder parents dropdown
* travis fixes

### CODE REVIEW

**Comment by Flight** [comment]
Some tests are failing here (problems with travis), will fix it together with @mikek soon. Gulp is ok on local machine.

File: filer/private/sass/components/_action-list.scss [hunk 1]
```diff
@@ -34,6 +34,7 @@
         display: block;
         padding: 5px 10px;
         border-bottom: solid 1px $gray-lighter;
+        font-size: 20px;
```
Comment (lory87) [inline filer/private/sass/components/_action-list.scss]: check order

File: filer/private/sass/components/_navigator.scss [hunk 2]
```diff
@@ -72,6 +80,14 @@
     background: $gray-table;
     .breadcrumbs-container {
         float: left;
+        width: calc(100% - 615px);
```
Comment (vxsx) [inline filer/private/sass/components/_navigator.scss]: what's 615 in this case? can it be a variable?

File: filer/templates/admin/filer/folder/directory_listing.html [hunk 3]
```diff
@@ -95,26 +95,33 @@
...
-                            title="{% trans 'Go back to the parent folder' %}">
-                            {{ folder.parent.name }}
-                        </a>
-                    {% endif %}
+                    <ul class="dropdown-menu navigator-breadcrumbs-dropdown">
+                        <li>
+                            <a href="{% url 'admin:filer-directory_listing-root' %}{% if is_popup %}?_popup=1{% if select_folder %}&amp;select_folder=1{% endif %}{% endif %}"
+                                title="{% trans 'Go back to' %} {% trans 'root'|title %} {% trans 'folder' %}">
+                                <img src="{% static 'filer/icons/root_28x28.png' %}" alt="{% trans 'Folder Icon' %}" width="28" height="28">
+                                Root
```
Comment (vxsx) [inline filer/templates/admin/filer/folder/directory_listing.html]: shouldn't it be translated?

File: filer/templates/admin/filer/folder/directory_listing.html [hunk 4]
```diff
@@ -128,18 +135,30 @@
...
                 {% endif %}
             </div>
         </div>
 
+        <form class="filter-files-container js-filter-files-container" action="." method="get" class="js-filer-search-form">
+            {% if is_popup %}
+                <input type="hidden" name="_popup" value="1" />
+                {% if select_folder %}<input type="hidden" name="select_folder" value="1" />{% endif %}
+            {% endif %}
+            <input type="text" placeholder="Search Files..." class="filter-files-field js-filter-files" value="{{ search_string }}" size="40" name="q">
```
Comment (vxsx) [inline filer/templates/admin/filer/folder/directory_listing.html]: should the placeholder be translated?

File: filer/templates/admin/filer/folder/directory_listing.html [hunk 5]
```diff
@@ -128,18 +135,30 @@
...
+            <div class="dropdown-container dropdown-container-down">
+                <a href="#" data-toggle="dropdown" aria-expanded="false">
+                    <span class="fa fa-caret-down"></span>
+                </a>
+                <ul class="dropdown-menu dropdown-menu-checkboxes">
+                    <li>
+                        <label><input type="checkbox" id="limit_search_to_folder" name="limit_search_to_folder" {% if limit_search_to_folder %}checked="checked"{% endif %} title="{% trans 'Check it to limit the search to current folder' %}"> limit the search to current folder</label>
```
Comment (vxsx) [inline filer/templates/admin/filer/folder/directory_listing.html]: here as well

File: filer/private/sass/components/_navigator.scss (hunk 2 above)
Comment (Flight) [inline filer/private/sass/components/_navigator.scss]: Nope, not here =(

File: filer/templates/admin/filer/folder/directory_listing.html (hunk 3 above)
Comment (Flight) [inline filer/templates/admin/filer/folder/directory_listing.html]: No, we decided not to translate it

**Comment by lory87** [comment]
:+1:

**Comment by vxsx** [comment]
LGTM

**Pull request merged**